documentation/config_file_auth_migration.md

```diff
@@ -0,0 +1,267 @@
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow:
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+4. Next, set the [`allow`](#allow-allow-unauthenticated-and-deny) setting. The legacy rule used a `*` glob, which is also supported in HOCON.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: *
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+5. Finally, give the rule a unique [`name`](#name) value. Remember that the rule will appear in logs and in the body of error responses to unauthorized clients.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: *
+            sort-order: 1
+            name: "environments"
+        },
+    ...
+    ```
+
+Our HOCON `auth.conf` file should now allow all clients to GET and POST to the `/puppet/v3/environments` endpoint, and should look like this:
+
+``` hocon
+authorization: {
+    version: 1
+    rules: [
+        {
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: *
+            sort-order: 1
+            name: "environments"
+        },
+    ]
+}
+```
+
+### Converting more complex rules
+
+#### Paths set by regular expressions
+
+To convert a regular expression path, enclose it in double quotation marks and solidus characters (`/`), and set the `type` to regex.
+
+> **Note:** You must escape regular expressions to conform to HOCON standards, which are the same as JSON's and differ from the deprecated format's regular expressions. For instance, the digit-matching regular expression `\d` must be escaped with a second backslash, as `\\d`.
+
+Deprecated:
+
+```
+path ~ ^/puppet/v3/catalog/([^/]+)$
+```
+
+HOCON:
+
+``` hocon
+...
+        match-request: {
+            path: "/^\/puppet\/v3\/catalog\/([^\/]+)$/"
+            type: regex
+...
+```
+
+> **Note:** You must escape regular expressions to conform to HOCON standards, which are the same as JSON's and differ from the deprecated format's regular expressions. For instance, the digit-matching regular expression `\d` must be escaped with a second backslash, as `\\d`.
+
+Backreferencing works the same way it does in the deprecated format.
+
+Deprecated:
+
+```
+path ~ ^/puppet/v3/catalog/([^/]+)$
+allow $1
+```
+
+HOCON:
+
+``` hocon
+...
+        match-request: {
+            path: "/^\/puppet\/v3\/catalog\/([^\/]+)$/"
+            type: regex
+        }
+        allow: $1
```

Comment:
I'm pretty sure you have to quote this, like ...

```
allow: "$1"
```
... since I think the `$` character is used as a marker for variable substitution in HOCON, which we wouldn't want in this example.  The above is what we do in the rules containing backreferences that we package by default in Puppet Server.